cldevice.go

```diff
@@ -601,10 +599,10 @@ func (d *Device) runDevice() error {
 
 	var status cl.CL_int
 	for {
-		d.updateCurrentWork()
+		d.updateCurrentWork(ctx)
 
 		select {
-		case <-d.quit:
+		case <-ctx.Done():
```

Comment:
These are inside tight mining loops and the call to `Done()` uses a mutex.  Would suggest the following to avoid needing the mutex inside the loop.  Same for the other `runDevice` cases.

```Go
ctxDoneCh := ctx.Done()
for {
	case <-ctxDoneCh:
```